test/controllers/PaypalControllerSpec.scala

```diff
@@ -74,23 +112,31 @@ class PaypalControllerSpec extends PlaySpec
   "Paypal Controller" should {
 
     "generate correct redirect URL for successful PayPal payments" in {
-      Mockito.when(mockPaypalService.executePayment(Matchers.anyString, Matchers.anyString)(Matchers.any[LoggingTags]))
-        .thenReturn(EitherT.right[Future, String, Payment](Future.successful(mockPaypalPayment)))
+      val fixture = new PaypalControllerFixture {
+        Mockito.when(mockPaypalService.executePayment(Matchers.anyString, Matchers.anyString)(Matchers.any[LoggingTags]))
+          .thenReturn(EitherT.pure[Future, String, Payment](mockPaypalPayment))
+      }
 
-      val result: Future[Result] = executePayment(fakeRequest)
+      val result: Future[Result] = executePayment(fixture.controller)(fakeRequest)
 
       status(result).mustBe(303)
       redirectLocation(result).mustBe(Some("/uk/post-payment"))
+
+      fixture.storeMetaDataCalled(1)
```

Comment:
It was quite non-obvious to me at first that this is another condition for the test to pass. It would be much clearer to me if it read something like

```scala
wasMetaDataStored().mustBe(true)
```

Is it easy to refactor it so it reads more like that?